src/python/pants/backend/python/goals/tailor.py

```diff
@@ -153,6 +178,13 @@ async def find_putative_targets(
         )
         pts.extend(source_targets)
 
+    if python_setup.tailor_py_typed_targets:
+        all_py_typed_files_globs: PathGlobs = req.path_globs("py.typed")
```

Comment:
Why create the glob outside the rule_helper instead of inside it?

Reply:
Oh I thought we want to do this since the `rule_helper` takes `PathGlobs` as it doesn't really need `PutativePythonTargetsRequest`, I reckon.

```
@rule_helper
async def _find_source_targets(
    py_files_globs: PathGlobs, all_owned_sources: AllOwnedSources, python_setup: PythonSetup
) -> list[PutativeTarget]:
```

This is how it's done above the new code:

```
    all_py_files_globs: PathGlobs = req.path_globs("*.py", "*.pyi")

    if python_setup.tailor_source_targets:
        source_targets = await _find_source_targets(
            all_py_files_globs, all_owned_sources, python_setup
        )
```

as well.

Reply:
Oh, makes sense.